pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/kits/KitRequestShippingTest.java

```diff
@@ -197,4 +203,29 @@ public void testPepperParticipantKitUpload() {
         });
     }
 
+    @Test
+    public void testNotHruidParticipantKitUpload() {
+        Profile mimicNotHruidParticipant = new Profile();
+        mimicNotHruidParticipant.setHruid(notHruidParticipantShortID);
+        notHruidParticipant = TestParticipantUtil.createParticipantWithEsProfile(notHruidParticipantGuid, mimicNotHruidParticipant,
+                ddpInstanceDto);
+        notHruidParticipantGuid = notHruidParticipant.getRequiredDdpParticipantId();
+        participants.add(notHruidParticipant);
+
+        String collaboratorParticipantId = "PROJ_" + notHruidId;
+        String collaboratorSampleId = collaboratorParticipantId + "_SALIVA";
+        // Checking the generated collaborator participant ID and sample ID for the case where the ID passed as short ID is not an HRUID
```

Comment:
Sorry, I'm still confused about what we are testing here.
What are we verifying by asserting that the collaborator participant and sample IDs created above are the same as those returned by the two methods below?

Reply:
No I am sorry, We are verifying that even if a participant has an HRUID, if the RGP subject ID (notHruidId) is passed as the short ID to this method, the returned collaborator participant ID and sample ID should use the subject ID (notHruidId) and not the HRUID. This ensures that the system handles the subject ID correctly when it is used for kit uploads, maintaining consistency with the participant's existing records.